suite/cstest/src/helper.c

```diff
@@ -0,0 +1,158 @@
+#include "helper.h"
+
+char **split(char *str, char *delim, int *size)
+{
+	char **result;
+	char *token, *src;
+	int cnt;
+	
+	cnt = 0;
+	src = str;
+	result = NULL;
+	while ((token = strstr(src, delim)) != NULL) {
+		result = (char **)realloc(result, sizeof(char *) * (cnt + 1));
+		result[cnt] = (char *)calloc(1, sizeof(char) * (int)(token - src + 10));
+		memcpy(result[cnt], src, token - src);
+		result[cnt][token - src] = '\0';
+		src = token + strlen(delim);
+		cnt ++;
+	}
+
+	if ( strlen(src) > 0 ) {
+		result = (char **)realloc(result, sizeof(char *) * (cnt + 1));
+		result[cnt] = strdup(src);
+		cnt ++;
+	}
+	*size = cnt;
+	return result;
+}
+
+void print_strs(char **list_str, int size)
+{
+	int i;
+	
+	printf("[+] Debug %d strings:\n", size);
+	for (i=0; i<size; ++i)
+		printf("String %d'th: %s\n", i+1, list_str[i]);
+}
+
+void free_strs(char **list_str, int size)
+{
+	int i;
+	for (i=0; i<size; ++i)
+		free(list_str[i]);
+	free(list_str);
+}
+
+const char *get_filename_ext(const char *filename)
+{
+	const char *dot;
+
+	dot = strrchr(filename, '.');
+	if(!dot || dot == filename)
+		return "";
+	return dot + 1;
+}
+
+char *readfile(const char *filename)
+{
+	char *result;
+	FILE *fp;
+	int size;
+
+	fp = fopen(filename, "r");
+	if (fp == NULL) {
+		puts("No such file");
+		exit(-1);
+	}
+	
+	fseek(fp, 0L, SEEK_END);
+	size = ftell(fp);
+	rewind(fp);
+
+	result = (char *)malloc(sizeof(char) * size);
+	fread(result, size, 1, fp);
+	
+	fclose(fp);
+	return result;
+}
+
+void addStr(char **src, const char *format, ...)
```

Comment:
to be consistent with your naming scheme, is it better to rename this to `add_str()`, or something?

same for other functions.